File: project4/main.cpp
#include <iostream> // заголовковий файл з потоками введення/виведення

  using namespace std; // використання простору імен std

void setArray(int M[]){ // функція для задання масиву
  for(int i = 0; i < 5; i++){
    cout << "Enter digit: ";
    cin >> M[i];
  }
}

void printArray(int M[], int size){ // функція для виведення масиву
  cout << endl;
  for(int i = 0; i < size; i++)
    cout << M[i] << '\t';
  cout << endl;
}

int Sum(int a, int b) { // розрахунок суми
  return a+b;
}

int Sum(int M1[], int M2[]){ // функція для поелементної суми масивів
  int sum;
  for(int i = 0; i < 5; i++)
    sum += M1[i] + M2[i];
  return sum; 
}

void BubbleSort(int M[], int size){ // сортування бульбашкою
  for (int i = 0; i < size - 1; i++) // проходи
    for (int j = 0; j < size - i - 1; j++) // ітерації сортування
      if (M[j] > M[j + 1]) 
        swap(M[j], M[j + 1]); // поміняту елементи місцями
  printArray(M, 5); // виклик функції для друку масиву
}

void InsertElement(int M[], int index, int value){ // функція для вставки елемента в масив
  for(int i = sizeof(M) / sizeof(M[0]) - 1; i > index; i--)
    M[i] = M[i-1]; // зсув/перестановка
  M[index] = value;
}

int main(){
  // оголошення масивів
  int M1[6], M2[6];
  // задання масивів через фунцкції
  setArray(M1);
  setArray(M2);
  // виведення ретексту і поверненого значення функції Sum
  cout << "Sum is: " << Sum(M1, M2) << endl;
  // Сортування масивів
  cout << "Array M1 now sorted: ";
  BubbleSort(M1, 5);
  cout << "Array M2 now sorted: ";
  BubbleSort(M2, 5);
  // вставка елемента в початок масиву
  cout << "Enter value to insert into array: ";
  int value;
  cin >> value;
  InsertElement(M1, 0, value);
  // виведення масиву
  cout << "Inserted element " << value << " in array M1: ";
  printArray(M1, 6);
  return 0;
}

File: project4/main.py
# функція для задання масиву
def setArray(M):
    for i in range(5):
        M[i] = int(input("Enter digit: "))

# сортування бульбашкою
def BubbleSort(M, size):
    for i in range(size - 1): # проходи
        for j in range(size - i - 1): # ітерації сортування
            if M[j] > M[j + 1]: 
                M[j], M[j + 1] = M[j + 1], M[j] # перестановка елементів місцями
    print(M)

# ініціалізація масивів нулями, на 5 елементів
M1 = [0] * 5
M2 = [0] * 5

# задання масивів через фунцкції
setArray(M1)
setArray(M2)

# виведення тексту і суми елементів списків
print(f"Sum is: {sum(M1 + M2)}")

print(" dadawj dnadkjawn dankwd nawdnwanjkdajkn\t",)
# Сортування масивів
print("Array M1 now sorted: ")
BubbleSort(M1, 5)
print("Array M2 now sorted: ")
BubbleSort(M2, 5)

# отрумання value
value = int(input("Enter value to insert into array: "))
# Вставка value по індексу 0
M1.insert(0, value)

# виведення масиву
print(f"Inserted element {value} in array M1: ")
print(M1)
